src/main/io/osd.c

```diff
@@ -672,21 +672,32 @@ static bool osdDrawSingleElement(uint8_t item)
             }
 
 #ifdef USE_ESC_SENSOR
-            // Show warning if we lose motor output
-            if (enabledWarnings & OSD_WARNING_ESC_FAIL && ARMING_FLAG(ARMED)) {
+            // Show warning if we lose motor output or the ESC is overheating
+            if (feature(FEATURE_ESC_SENSOR) && enabledWarnings & OSD_WARNING_ESC_FAIL) {
                 char escErrMsg[OSD_FORMAT_MESSAGE_BUFFER_SIZE];
-                // center justify message
                 int pos = 0;
-                while (pos < 4 - getMotorCount()/2) {
+
+                // center justify message
+                while (pos < 4 - getMotorCount() / 2) {
```

Comment:
This needs to be updated to pos < (OSD_FORMAT_MESSAGE_BUFFER_SIZE - strlen('ESC') - getMotorCount())/2

Reply:
Good point.